src/http-api.js

```diff
@@ -389,6 +389,10 @@ module.exports.MatrixHttpApi.prototype = {
         prefix,
         accessToken,
     ) {
+        if (!this.opts.idBaseUrl) {
```

Comment:
We should also guard:

* `MatrixBaseApis.prototype.registerWithIdentityServer`

Also not quite sure what this one will do for empty string:

* `MatrixClient.prototype._requestTokenFromEndpoint`

Reply:
url.parse will parse the empty string to null everything so the second one if fine as-is